### DISCUSSION TITLE
Provide tooling for Trust policy modification

### DISCUSSION THREAD

**Discussion by sajayantony** [notaryproject/notation#285] [Ideas]
The current trust policy json is hard to validate/test due to the lack of tooling experience in modifying/creating a baseline. 
I believe there are couple of options for providing a better user experience. 

- Option 1 : CLI Based  
    - This would be a minimum bar and I believe we can include things like even template generation on the CLI. 
        - For e.g. `notation policy ...` 
- Option 2 : UI based. 
    - This makes editing much easier and there might be an opportunity for notary project to provide an extension on VSCode that can even be customized by various environments through extension profiles. 
    - For e.g. disable certain external registries or repositories. 

For VSCode we can consider the WEBUI Framework https://code.visualstudio.com/blogs/2021/10/11/webview-ui-toolkit
This would also help create a wizard that can provide policy editing. 

I see that there might be challenges in building this out given the priorities of the project. But I feel its worth considering since the experience hasn't yet been finalized. 

+ @FeynmanZhou

**Comment by FeynmanZhou** [comment]
I agree with option 1 of providing CLI compatibility for editing trust policy and generating trust policy template then allowing users to edit it accordingly. There was [an issue](https://github.com/notaryproject/roadmap/issues/38) tracking the CLI design which can be implemented post RC.1. We can start to work on its CLI design soon.

Option 2 is a good proposal for newbies to edit the trust policy with the wizard UI, which brings a hassle-free user experience. We can open this for gathering users' feedback and determine whether it can be supported post Option1 implementation.
